examples/design/EU-DEMO/EUDEMO_builders/ivc/wall_silhouette.py

```diff
@@ -0,0 +1,381 @@
+        """
+        Optimise the shape using the provided parameterisation and optimiser.
+        """
+        if not hasattr(self, "problem_class"):
+            raise ValueError(
+                f"Cannot execute {type(self).__name__} in RUN mode: no problem_class specified."
+            )
+
+        bluemira_debug(
+            f"Setting up design problem with:\n"
+            f"algorithm_name: {self.algorithm_name}\n"
+            f"n_variables: {parameterisation.variables.n_free_variables}\n"
+            f"opt_conditions: {self.opt_conditions}\n"
+            f"opt_parameters: {self.opt_parameters}"
+        )
+
+        optimiser = Optimiser(
+            self.algorithm_name,
+            parameterisation.variables.n_free_variables,
+            self.opt_conditions,
+            self.opt_parameters,
+        )
+
+        if self._problem_settings != {}:
+            bluemira_debug(
+                f"Applying non-default settings to problem: {self.problem_settings}"
+            )
+        design_problem = self.problem_class(
+            parameterisation,
+            optimiser,
+            self._make_wall_keep_out_zone(),
+            **self.problem_class_kwargs,
+            **self._problem_settings,
+        )
+
+        bluemira_print(f"Solving design problem: {design_problem.__class__.__name__}")
+        if parameterisation.n_ineq_constraints > 0:
+            bluemira_debug("Applying shape constraints")
+            design_problem.apply_shape_constraints()
+
+        bluemira_debug("Solving...")
+        return design_problem.optimise()
+
+    def _make_wall_keep_out_zone(self) -> BluemiraWire:
+        """
+        Create a "keep-out zone" to be used as a constraint in the
+        wall shape optimiser.
+        """
+        geom_offset = self.params.tk_sol_ib.value
+        psi_n = self.params.fw_psi_n.value
+        geom_offset = 0.2  # TODO: Unpin
+        psi_n = 1.05  # TODO: Unpin
+        geom_offset_zone = self._make_geometric_keep_out_zone(geom_offset)
+        flux_surface_zone = self._make_flux_surface_keep_out_zone(psi_n)
+        leg_zone = self._make_divertor_leg_keep_out_zone(
+            self.params.div_L2D_ib.value, self.params.div_L2D_ob.value
+        )
+        return convex_hull_wires_2d(
+            [geom_offset_zone, flux_surface_zone, leg_zone], ndiscr=200, plane="xz"
+        )
+
+    def _make_geometric_keep_out_zone(self, offset: float) -> BluemiraWire:
+        """
+        Make a "keep-out zone" from a geometric offset of the LCFS.
+        """
+        lcfs = make_polygon(self.equilibrium.get_LCFS().xyz, closed=True)
+        return offset_wire(lcfs, offset, join="arc")
+
+    def _make_flux_surface_keep_out_zone(self, psi_n: float) -> BluemiraWire:
+        """
+        Make a "keep-out zone" from an equilibrium's flux surface.
+        """
+        # TODO: This is currently called twice once here and once for the
+        # setup of the remaining ivc
+        o_points, _ = find_OX_points(
```

Comment:
possibly also from plasma componentmanager

Reply:
also for comment context see [WIP reactor](https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/5c93cd6a22e8979b0bfa4f91102a455613f92f05/examples/design/EU-DEMO/EUDEMO_reactor_WIP.py)